crates/oci-distribution/src/client.rs

```diff
@@ -751,12 +753,88 @@ impl ClientProtocol {
+    /// Custom deserialize implementation for RegistryToken.
+    ///
+    /// When requesting a scoped authorization token from a registry,
+    /// an opaque token (typically JWT) is returned in a JSON response.
+    /// Some registries place this token in the 'token' field, whereas
+    /// other registries place this token in the 'access_token' field.
+    /// In rare cases, both 'token' and 'access_token' are provided
+    /// with identical values for compatibility purposes. In this
+    /// scenario, the 'access_token' field can be safely ignored.
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct RegistryTokenVisitor;
+
+        impl<'de> de::Visitor<'de> for RegistryTokenVisitor {
+            type Value = RegistryToken;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("RegistryToken")
+            }
+
+            fn visit_map<V>(self, mut map: V) -> Result<Self::Value, V::Error>
+            where
+                V: de::MapAccess<'de>,
+            {
+                let mut token = String::new();
+                loop {
+                    let next_key_result = map.next_key::<String>();
+                    if next_key_result.is_err() {
+                        return Err(de::Error::custom("Unable to parse JSON"));
+                    }
+                    let next_key_option = next_key_result.unwrap();
+                    if next_key_option.is_none() {
+                        break; // done reading the map
+                    }
+                    let next_value_result = map.next_value::<String>();
+                    if next_value_result.is_err() {
+                        continue; // does not parse as string-to-string (e.g. 'expires_in' field)
+                    }
+                    let key = next_key_option.unwrap();
+                    let val = next_value_result.unwrap();
+                    let is_primary_field = key == REGISTRY_TOKEN_JSON_PRIMARY_FIELD;
+                    let is_secondary_field = key == REGISTRY_TOKEN_JSON_SECONDARY_FIELD;
+                    if is_primary_field || is_secondary_field {
+                        if !token.is_empty() {
+                            debug!(
+                                "Warning: Auth response JSON body contained both a '{}' field \
+                                and a '{}' field (ignoring the latter)",
+                                REGISTRY_TOKEN_JSON_PRIMARY_FIELD,
+                                REGISTRY_TOKEN_JSON_SECONDARY_FIELD
+                            );
+                            if is_secondary_field {
+                                continue; // do not overwrite value from primary with secondary
+                            }
+                        }
+                        token = val;
+                    }
+                }
+                if token.is_empty() {
```

Comment:
Are you using the empty string for "we haven't found a token"?  If so, would it be appropriate to use an `Option` instead?  Or is there a potential case where you read the token and it's empty?